linode/manager
# upcoming: [DI-18419] - API error handling for global filters and dashboard component in ACLP

## Description 📝
Added code for handling API errors in global filters and dashboard component

## Changes  🔄

1. Added code for handling API errors in global filters
2. Added error handling code in dashboard component
3. Added default values for widget color and chart type

## Target release date 🗓️
11-11-2024

## Preview 📷
| Updates  | Updates |
| ------- | ------- |
|![Screenshot 2024-10-28 at 4 34 23 PM](https://github.com/user-attachments/assets/42440a5b-90e0-46ab-ae28-7fe494573827)|![Screenshot 2024-10-28 at 4 35 52 PM](https://github.com/user-attachments/assets/afbfb766-9f6e-407f-9fb0-826a9f650b68)|
|![Screenshot 2024-10-28 at 4 36 30 PM](https://github.com/user-attachments/assets/40ade22b-4381-4af6-b645-717e1aa5d060)|![Screenshot 2024-10-28 at 4 37 20 PM](https://github.com/user-attachments/assets/f7caf02c-04fe-4b01-86b2-31b1762cbcc3)|
|![Screenshot 2024-10-28 at 4 38 51 PM](https://github.com/user-attachments/assets/5609d8ae-0c2b-466e-af61-1a53fffdbf3e)|![Screenshot 2024-10-28 at 4 40 02 PM](https://github.com/user-attachments/assets/9d1106b1-6b1a-4baa-87f7-ea93ce8b4627)|

## How to test 🧪

1. Login as mock user as some of the endpoints are not enabled in production
2. Open inspect screen
3. Navigate to monitor tab
4. You can use the serverHandler.ts where our mocks are present, in each of the API's
 **(monitor/services, monitor/services/:serviceType/dashboards, /regions, linode/instances in case of linode dashboard, databases/instances incase of dbaas dashboard) instead of returning a valid response, you can return HttpResponse.error()** and see the error messages one by one.

## As an Author I have considered 🤔

*Check all that apply*
